actionmailer/lib/action_mailer/structured_event_subscriber.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require "active_support/structured_event_subscriber"
+
+module ActionMailer
+  class StructuredEventSubscriber < ActiveSupport::StructuredEventSubscriber # :nodoc:
+    # An email was delivered.
+    def deliver(event)
+      if (exception = event.payload[:exception_object])
+        emit_debug_event("action_mailer.delivery_error",
```

Comment:
Why are some events debug events and others not?

Reply:
This is matching the debug log level events in the corresponding log subscribers. Debug level is essentially development, so we don't want these events emitted in test and prod.